modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs

```diff
@@ -427,10 +427,14 @@ static void LookupScriptForClass(Type type)
             // This method may be called before initialization.
             if (NativeFuncs.godotsharp_dotnet_module_is_initialized().ToBool() && Engine.IsEditorHint())
             {
-                foreach (var scriptPath in _pathTypeBiMap.Paths)
+
+                if (_pathTypeBiMap.Paths.Count > 0)
                 {
-                    using godot_string nativeScriptPath = Marshaling.ConvertStringToNative(scriptPath);
-                    NativeFuncs.godotsharp_internal_editor_file_system_update_file(nativeScriptPath);
+                    string[] script_paths = _pathTypeBiMap.Paths.ToArray();
+                    using (godot_packed_string_array script_paths_packed_array = Marshaling.ConvertSystemArrayToNativePackedStringArray(script_paths))
+                    {
+                        NativeFuncs.godotsharp_internal_editor_file_system_update_files(script_paths_packed_array);
+                    }
```

Comment:
No need to use braces here since the scope is already delimited by the if.

We also use the `Native` suffix to refer to variables of an interop type. So in this case, the managed variable `scriptsPath` is converted to an unmanaged type and stored in a variable named `scriptsPathNative` that can be used with the NativeFuncs.

```suggestion
                    using godot_packed_string_array scriptPathsNative = Marshaling.ConvertSystemArrayToNativePackedStringArray(scriptPaths);
                    NativeFuncs.godotsharp_internal_editor_file_system_update_files(scriptPathsNative);
```